src/main/java/org/opensearch/sdk/SDKClient.java

```diff
@@ -219,15 +227,14 @@ public OpenSearchAsyncClient initalizeJavaAsyncClient(String hostAddress, int po
      * <p>
      * Do not use this client for new development.
      *
-     * @param settings The Extension settings
      * @return The SDKClient implementation of RestHighLevelClient. The user is responsible for calling
      *         {@link #doCloseHighLevelClient()} when finished with the client
      * @deprecated Provided for compatibility with existing plugins to permit migration. Use
      *             {@link #initializeJavaClient} for new development.
      */
     @Deprecated
-    public SDKRestClient initializeRestClient(ExtensionSettings settings) {
-        return initializeRestClient(settings.getOpensearchAddress(), Integer.parseInt(settings.getOpensearchPort()));
+    public SDKRestClient initializeRestClient() {
+        return initializeRestClient(extensionSettings.getOpensearchAddress(), Integer.parseInt(extensionSettings.getOpensearchPort()));
```

Comment:
These changes will break AD extension. @Kuanysh-kst can you incorporate the changes [here](https://github.com/opensearch-project/anomaly-detection/blob/feature/extensions/src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java#L187) and [here](https://github.com/opensearch-project/anomaly-detection/blob/feature/extensions/src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java#L197).

Make sure to raise your PR on AD plugin repo against `feature/extensions` branch and not `main`. Thanks

Reply:
> These changes will break AD extension. @Kuanysh-kst can you incorporate the changes [here](https://github.com/opensearch-project/anomaly-detection/blob/feature/extensions/src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java#L187) and [here](https://github.com/opensearch-project/anomaly-detection/blob/feature/extensions/src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java#L197)?

Actually that first one (line 187) will still work as it extracts the host and port.  The second one (line 197) would break; but it could be switched to the first format temporarily without breaking anything, and then both switched to the simpler version after this is merged.

To be more clear, are you OK with this PR but want to have a PR to fix AD ready to merge as soon as this one goes in?

Reply:
Yes. That would work

Reply:
Hey @owaiskazi19 seems AD Extension is already broken with the merged change of `BaseExtension` inheritance.  So it wouldn't hurt to merge this PR and break it more. :)

I've already created https://github.com/opensearch-project/anomaly-detection/pull/843 to fix both problems.